ambry-rest/src/main/java/com.github.ambry.rest/Http2ChannelPoolMap.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.rest;
+
+import com.github.ambry.config.SSLConfig;
+import com.github.ambry.network.RequestInfo;
+import io.netty.bootstrap.Bootstrap;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.pool.AbstractChannelPoolMap;
+import io.netty.channel.pool.ChannelPool;
+import io.netty.channel.socket.nio.NioSocketChannel;
+
+
+public class Http2ChannelPoolMap extends AbstractChannelPoolMap<RequestInfo, ChannelPool> {
```

Comment:
It seems that RequestInfo isn't exactly the right key to use for `Http2ChannelPoolMap` and `Http2ChannelInitializer` for a couple reasons:
1. It doesn't have hashcode and equals defined, and a separate object is created for each request, so a new pool will created for each request, instead of each host:port combo
2. It includes info about the request body, which is not relevant to the pool (i think).

Instead, you could use `InetSocketAddress` and pass this directly into `Bootstrap::remoteAddress`. The impl of `Bootstrap::remoteAddress(String inetHost, int inetPort)` would call `InetSocketAddress.createUnresolved(inetHost, inetPort)` anyways.

Reply:
InetSocketAddress is used.